galaxyui/src/app/resources/respositories/repository.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable }              from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+import { Observable }           from 'rxjs/Observable';
+import { of }                   from 'rxjs/observable/of';
+import { catchError, map, tap } from 'rxjs/operators';
+
+import { Repository }          from './repository';
+import { NotificationService } from 'patternfly-ng/notification/notification-service/notification.service'
+import { QueryResponse }       from "../query-response";
+
+const httpOptions = {
+    headers: new HttpHeaders({
+        'Content-Type': 'application/json'
+    })
+};
+
+@Injectable()
+export class RepositoryService {
+
+    private url = '/api/v1/repositories';
+
+    constructor(private http: HttpClient,
+                private notificationService: NotificationService) {
+    }
+
+    query(): Observable<Repository[]> {
+        return this.http.get<QueryResponse>(this.url)
+            .pipe(
+                map(response => response.results),
+                tap(_ => this.log('fetched repositories')),
+                catchError(this.handleError('Query', []))
+            );
+    }
+
+    save(repository: Repository): Observable<Repository> {
+        console.log('RepositoryService.save', this.url, repository, httpOptions);
+        return this.http.post<Repository>(this.url, repository, httpOptions).pipe(
```

Comment:
@chouseknecht this http.post is failing to execute and I've stared at it too long and can't see the issue.  I need another pair of eyes.

Reply:
@chouseknecht figured it out.  There was nothing wrong with the service.  I didn't do a .subscribe when calling the .save from the modal dialog so the http.post was never executed.